### PR TITLE
Show warnings

### LINKED ISSUE #26039: `system.warnings` table and warnings in `clickhouse-client`
The server should have a shared array of warning messages.
You can put it simply as `std::vector<std::string>` in `ContextShared`.

There should be a method to add a new warning.
This method will print a log message and also add the warning to the list.
Warnings are not persisted (the list is empty on server startup and will be filled at runtime).

A system table `system.warnings` should be created. Reading from this table will return all the warnings.
(See `IStorageSystemOneBlock` and its usages). The table structure is one column `message String`.

When `clickhouse-client` is connected to the server in interactive mode, it should do `SELECT message FROM system.warnings` (simple synchronous query) and if there are warnings, print them in form of `Warning: message...` each on separate line. If the table does not exist on server or if query cannot execute by another reason, catch exception and ignore it.

It will be used for #18794
Example: if server was build in debug mode, print a warning about it in client.

PS. Not every log message with warning level goes to `system.warnings`. The `system.warnings` is only for special kinds of warnings that we are going to collect.

### PR DESCRIPTION
**Pull request by FArthur-cmd**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category:
- Improvement

Changelog entry:
- When client connect to server, he receives information about all warnings that are already were collected by server. (It can be disabled by using option `--no-warnings`)

Detailed description / Documentation draft:

- When client connect to server, he sees all warnings. That may be useful for understanding slowness of work (for example, because of debug mode) or other parameters, that can influence on work. 
- Add option `--no-warnings` for client. It turns off printing warnings which are shown after connection (you can still see all warnings using `SELECT message FROM system.warnings`).

### CODE REVIEW

File: programs/client/Client.cpp [hunk 1]
```diff
@@ -564,6 +610,25 @@ class Client : public Poco::Util::Application
...
             }
+            /// Load Warnings at the begining of connection
+            {
+                std::vector<String> messages;
+                try
+                {
+                    loadWarningMessages(messages);
+                    if (!messages.empty())
+                    {
+                        std::cout << "Warnings:" << std::endl;
```
[the remark this answers is not in the document]
Comment (FArthur-cmd) [reply programs/client/Client.cpp:621]: "When clickhouse-client is connected to the server in interactive mode, it should do SELECT message FROM system.warnings (simple synchronous query) and if there are warnings, print them in form of Warning: message... each on separate line. If the table does not exist on server or if query cannot execute by another reason, catch exception and ignore it."

As I understand, user should see these warnings

**Comment by FArthur-cmd** [comment]
Test 01526_client_start_and_exit in Functional stateless tests flaky check (address) works correctly, but slow. Probably, it can be ignored. Also check in Integration tests is flaky, so we can skip it.

File: programs/client/Client.cpp [hunk 2]
```diff
@@ -487,6 +488,51 @@ class Client : public Poco::Util::Application
     }
 #endif
 
+    /// Make query to get all server warnings
+    void loadWarningMessages(std::vector<String>& messages)
+    {
+        connection->sendQuery(connection_parameters.timeouts, "SELECT message FROM system.warnings", "" /* query_id */, QueryProcessingStage::Complete);
```
Comment (alexey-milovidov) [inline programs/client/Client.cpp:494]: > If the table does not exist on server or if query cannot execute by another reason, catch exception and ignore it.

It is for compatibility with old servers.

**Comment by alexey-milovidov** [comment]
Now we can add a test with `expect`.

PS. Note that the usage of `expect` was unneeded in #26246.

File: programs/client/Client.cpp (hunk 2 above)
Comment (FArthur-cmd) [reply programs/client/Client.cpp:494]: Is it bad to have try-catch over all this function? (lines 618-632)

Comment (alexey-milovidov) [reply programs/client/Client.cpp:494]: It is alright.

**Review by alexey-milovidov — APPROVED** [review]

**Comment by alexey-milovidov** [comment]
Unfortunately one issue has been found after merge.

If no warnings, there was ugly excessive empty line:
```
ClickHouse client version 21.9.1.7510 (official build).
Connecting to localhost:9000 as user default.
Connected to ClickHouse server version 21.9.1 revision 54449.


ip-172-31-7-191.us-east-2.compute.internal :)
```

Should be:
```
```
ClickHouse client version 21.9.1.7510 (official build).
Connecting to localhost:9000 as user default.
Connected to ClickHouse server version 21.9.1 revision 54449.

ip-172-31-7-191.us-east-2.compute.internal :)
```
```

**Comment by alexey-milovidov** [comment]
Fixed in 2c3c329.

**Pull request merged**